internal/util/validation/quadlet.go

```diff
@@ -0,0 +1,119 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/flightctl/flightctl/internal/api/common"
+)
+
+// ValidateQuadletSpec verifies the QuadletSpec for common issues
+func ValidateQuadletSpec(spec *common.QuadletSpec) []error {
```

Comment:
We should be also validating the "Header" in the files.

eg. .container files should have a `[Container]` header, etc.